pkg/controller/build/reconciler.go

```diff
@@ -483,7 +482,9 @@ func (b *buildReconciler) getMachineOSBuildStatusForBuilder(ctx context.Context,
 		return mcfgv1alpha1.MachineOSBuildStatus{}, nil, fmt.Errorf("could not get MachineOSConfig or MachineOSBuild for builder: %w", err)
 	}
 
-	observer := imagebuilder.NewPodImageBuildObserverFromBuilder(b.kubeclient, b.mcfgclient, mosb, mosc, builder)
+	// TODO: Maybe infer MachineOSBuild state from the provided build object
```

Comment:
note (non-blocking): This is what the `...FromBuilder` constructor should do. In other words, it uses the object provided from the informer with its current state.